sdk/storage_datalake/src/operations/file_system_create.rs

```diff
@@ -0,0 +1,97 @@
+use crate::clients::FileSystemClient;
+use crate::util::*;
+use crate::Properties;
+use azure_core::prelude::*;
+use azure_core::{
+    collect_pinned_stream,
+    headers::{add_mandatory_header2, add_optional_header2},
+    AppendToUrlQuery, Response as HttpResponse,
+};
+use azure_core::{
+    headers::{etag_from_headers, last_modified_from_headers},
+    Etag,
+};
+use azure_storage::core::headers::CommonStorageResponseHeaders;
+use chrono::{DateTime, Utc};
+use std::convert::TryInto;
+
+/// A future of a create file system response
+type CreateFileSystem =
+    futures::future::BoxFuture<'static, crate::Result<CreateFileSystemResponse>>;
+
+#[derive(Debug, Clone)]
+pub struct CreateFileSystemBuilder {
+    client: FileSystemClient,
+    client_request_id: Option<ClientRequestId>,
+    timeout: Option<Timeout>,
+    properties: Option<Properties>,
+}
+
+impl CreateFileSystemBuilder {
+    pub(crate) fn new(client: FileSystemClient) -> Self {
+        Self {
+            client,
+            client_request_id: None,
+            timeout: None,
+            properties: None,
+        }
+    }
+
+    setters! {
+        client_request_id: ClientRequestId => Some(client_request_id),
+        timeout: Timeout => Some(timeout),
+        properties: Properties => Some(properties),
+    }
+
+    pub fn into_future(self) -> CreateFileSystem {
+        let this = self.clone();
+        let ctx = self.client.context.clone();
+
+        Box::pin(async move {
+            let mut url = this.client.url().clone();
```

Comment:
Does this have to be cloned?

Reply:
Right now we return a borrowed reference so we cannot make it mutable. This too will be gone in the next PR.